table_engine/src/partition/rule/df_adapter/extractor.rs

```diff
@@ -63,7 +63,35 @@ impl FilterExtractor for KeyExtractor {
                     (Expr::Column(col), Operator::Eq, Expr::Literal(val))
                     | (Expr::Literal(val), Operator::Eq, Expr::Column(col)) => {
                         let datum_opt = Datum::from_scalar_value(&val);
-                        datum_opt.map(|d| PartitionFilter::new(col.name, PartitionCondition::Eq(d)))
+                        datum_opt.map(|datum| {
+                            PartitionFilter::new(col.name, PartitionCondition::Eq(datum))
+                        })
+                    }
+                    _ => None,
+                },
+                Expr::InList {
+                    expr,
+                    list,
+                    negated,
+                } => match (*expr, list, negated) {
+                    // The 'not in' syntax is not currently supported.
+                    (Expr::Column(col), list, false) => {
```

Comment:
`list` is unneccessary to match.

Reply:
`list` is moved in  `match (*expr, list, negated)`, so we must declare it again.

Reply:
It seems strange?
How about:
```suggestion
{
      if  let (Expr::Column(col), false) = (Expr::Column(col), false) {
         ...
      } else {
           None
      }
 }
```